### DISCUSSION TITLE
Prevent jreleaserConfig from failing instantly

### DISCUSSION THREAD

**Discussion by crschnick** [jreleaser/jreleaser#1182] [Q&A]
I'm using jreleaser to release artifacts from multiple operating systems using GitHub actions by downloading the artifacts for the current action run. As a result, in the `distributions` block, there are many files listed that are only available altogether in a GitHub actions runner.
This causes the `jreleaserConfig` task to always tell me that an artifact is missing and fail instantly on my local machine.

It would be nice to have have an option to tell the task to either ignore missing distribution files or ignore configuration errors altogether and still print the rest of the configuration.

**Comment by aalmiray** [comment]
This is already possible. You can filter artifacts (include/exclude) by platform. See https://jreleaser.org/guide/latest/tools/jreleaser-gradle.html#_jreleaserconfig, specifically:

```
     --reject-platform          Activates paths not matching the given platform (OPTIONAL).

     --select-current-platform  Activates paths matching the current platform (OPTIONAL).

     --select-platform          Activates paths matching the given platform (OPTIONAL).
```

For example, listing all artifacts that you built locally for your current platform:

```
./gradlew jreleaserConfig --select-current-platform
```

Or listing all artifacts except `windows-x86_64` (assuming all other artifacts are available):

```
./gradlew jreleaserConfig --reject-platform windows-x86_64
```

FWIW many commands support these options as well.

**Reply by crschnick** [reply]
Oh, I even looked at the documentation you linked but did not associate these command-line arguments to the distribution paths. Because from an outsider's perspective, these descriptions are very vague.

**Reply by aalmiray** [reply]
Right. I'll add specific examples of these flags to the docs.

**Reply by aalmiray** [reply]
Added https://github.com/jreleaser/jreleaser.github.io/commit/3042f73a2d89872687b197aaec4bf9489aedbd22

**Reply by crschnick** [reply]
Thanks, that is much clearer now

**Comment by crschnick** [comment]
But the general issue of the `jreleaserConfig` task failing instantly has also bothered me in a couple of other cases, which are unrelated to distribution files. For example, any tokens that are only available in GitHub actions will also cause the task to fail. Maybe there could be an option like `--report-all` to report all issues at once?

**Reply by aalmiray** [reply]
Well, the tool already collects as many errors as possible. However, given the nature of convention over configuration, it errs on the side of caution and fails on the first critical misconfiguration because if it didn't the number of errors would increase dramatically. 

For example, run `gradlew jreleaserAnnounce` with a single announcer configured and no explicit project metadata. You'll likely get a couple of errors regarding project's url, license, copyright, and inception year. If you configure a jlink assembler without a `java` block and execute `gradlew jreleaserConfig --assembly` you should get at least one warning that the java block is missing but no warning about a missing release token (github in this case) if the token is not configured at all.

**Reply by aalmiray** [reply]
Additionally, there's a new `env` command ([gradle](https://jreleaser.org/guide/early-access/tools/jreleaser-gradle.html#_jreleaserenv)) that may be used to display environment variable configuration at any time. Also, the `trace.log` prints out which sys props and env vars (and from which location) were sourced when resolving the model. What's left to do is to print the exact source (sys props, env var, build tool, DSL) next to model property values as suggested at https://github.com/jreleaser/jreleaser/issues/1180#issuecomment-1402252220

Here's how the trace.log may look

```
[DEBUG] The following System properties are defined
[DEBUG]   jreleaser.distributions.active
[DEBUG] The following variables are defined in the current environment
[DEBUG]   JRELEASER_HOME
[DEBUG]   JRELEASER_PROJECT_VERSION
[INFO]  Loading variables from /Users/aalmiray/.jreleaser/config.toml
[DEBUG] The following variables are defined in config.toml
[DEBUG]   JRELEASER_ARTIFACTORY_KORDAMP_PASSWORD
[DEBUG]   JRELEASER_ARTIFACTORY_KORDAMP_USERNAME
[DEBUG]   JRELEASER_CHOCOLATEY_API_KEY
[DEBUG]   JRELEASER_CLOUDSMITH_TOKEN
[DEBUG]   JRELEASER_CODEBERG_TOKEN
...
```
